test/node/conversation-labels.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const assert = require('assert');
+const Circuit = require('../../circuit-node');
+const config = require('./config.json');
+const helper = require('./helper');
+Circuit.logger.setLevel(Circuit.Enums.LogLevel.Error);
+
+let client;
+let addedLabelsHT = {};
+describe('Authentication', () => {
+
+    it('should login using Client Credentials', async () => {
+        client = new Circuit.Client(config.bot1);
+        const user = await client.logon();
+        assert(!!user);
+    });
+
+    it('should be authenticated', async () => {
+        await client.isAuthenticated();
+    });
+
+
+    it('should add two labels', async () => {
+        const labelValue1 = Date.now().toString() + 'a';
+        const labelValue2 = Date.now().toString() + 'b';
+        const addedLabels = await client.addLabels([labelValue1, labelValue2]);
+        addedLabels.forEach(label => addedLabelsHT[label.labelId] = label);
+        const existingLabels = await client.getAllLabels();
+        const existingLabelsHT = {};
+        existingLabels.forEach(label => existingLabelsHT[label.labelId] = label); 
+        Object.keys(addedLabelsHT).forEach(testLabelId => {
+            if (!existingLabelsHT[testLabelId] || existingLabelsHT[testLabelId].value !== addedLabelsHT[testLabelId].value) {
+                // if the test label doesn't exist in the labels or if they don't equal each other fails the test
+                assert(false);
+            }
+        });
+    });
+
+    it('should edit the first added labels', async () => {
+        const labelIdToEdit = Object.keys(addedLabelsHT)[0];
```

Comment:
Properties order in objects is not guaranteed in JavaScript. Use an array or map.
See https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order